metagraph/src/graph/alignment/aligner_labeled.hpp

```diff
@@ -0,0 +1,278 @@
+#ifndef __LABELED_ALIGNER_HPP__
+#define __LABELED_ALIGNER_HPP__
+
+#include <optional>
+
+#include <tsl/hopscotch_map.h>
+#include <tsl/ordered_set.h>
+
+#include "dbg_aligner.hpp"
+#include "common/hashers/hash.hpp"
+#include "graph/annotated_dbg.hpp"
+#include "annotation/binary_matrix/base/binary_matrix.hpp"
+#include "annotation/int_matrix/base/int_matrix.hpp"
+
+namespace mtg {
+namespace graph {
+namespace align {
+
+
+template <typename Key, class Hash = std::hash<Key>, typename IndexType = uint64_t,
+          class EqualTo = std::equal_to<Key>, class Allocator = std::allocator<Key>,
+          class Container = std::vector<Key, Allocator>>
+using VectorSet = tsl::ordered_set<Key, Hash, EqualTo, Allocator, Container, IndexType>;
+
+
+class DynamicLabeledGraph {
+  public:
+    typedef DeBruijnGraph::node_index node_index;
+    typedef annot::binmat::BinaryMatrix::Column Column;
+    typedef annot::binmat::BinaryMatrix::Row Row;
+    typedef annot::matrix::MultiIntMatrix::Tuple Tuple;
+
+    typedef std::reference_wrapper<const Alignment::LabelSet> LabelSet;
+    typedef std::reference_wrapper<const Alignment::CoordinateSet> CoordsSet;
+
+    static constexpr Row nrow = std::numeric_limits<Row>::max();
+
+    DynamicLabeledGraph(const AnnotatedDBG &anno_graph);
```

Comment:
Why do you need this guy?

Reply:
Maybe I should call this `AnnotationBuffer` instead?

Reply:
maybe